### ISSUE TITLE
Certificate verification fails on https api call to GetFilesArchive()

### ISSUE THREAD

**Issue by soundwave01** [google/grr#569]
**What is the issue**

when using using the API over HTTPS the following code

grrapi.Client(client.client_id).File("/fs/tsk/").GetFilesArchive().WriteToFile(output_zip)

results in 

requests.exceptions.SSLError: ("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')],)",)

Other api calls including search for clients, starting flow, checking the status of a flow, all work without issue.  This is happening on version 3.2.1.post4 of grr-api-client installed via pip.

**Code to replicate the issue**

```
import grr_api_client
import os
import time

from grr_api_client import api

os.environ['REQUESTS_CA_BUNDLE'] = "/etc/ssl/certs/"

grrapi = grr_api_client.api.InitHttp(api_endpoint="https://xxxxxxxxxxxxxxxxxx:8000/",auth=("xxxxxxxx", "xxxxxxxxxx"))
target = "xxxxxxxxxxxxxxxx"
clients = grrapi.SearchClients("host:%s" % (target))

if not clients:
    print "[E]: Can't find host %s - Skipping" % (target)

# if we have multiple clients for the same host name
# get the most recent one
newestClient = None
counter = 0
for client in clients:

    if (counter == 0):
        newestClient = client
    if (client.data.last_seen_at > newestClient.data.last_seen_at):
        newestClient = client
    counter += 1

if (counter == 0):
    print "[E]: Can't find host %s - Skipping" % (target)
    #return

print "[+]: %s found" % (target)
client = newestClient

#flow args
flow_args = grrapi.types.CreateFlowArgs("ArtifactFilesDownloaderFlow")
flow_args.artifact_list.append("NTFSMFTFiles")
flow_args.artifact_list.append("WindowsSystemRegistryFiles")
flow_args.artifact_list.append("WindowsUserRegistryFiles")
flow_args.use_tsk = True
flow_args.max_file_size = 10000000000

#create the flow
flow = client.CreateFlow(name="ArtifactFilesDownloaderFlow", args=flow_args)


flowID = flow.flow_id
print "[+]: Flows started for %s" % (target)


timeout = 180
timeoutcounter = 0
flowWorked = True

#check the status of the flow
while(flow.data.state == flow.data.RUNNING):
    time.sleep(60)
    flow = grrapi.Client(client.client_id).Flow(flowID).Get()
    timeoutcounter += 1
    if(timeoutcounter == timeout):
        #Time out situation - cancel the thread
        print "[E] Flow for %s timed out - cancelling the flow" % (target)
        grrapi.Client(client.client_id).Flow(flowID).Get().Cancel()
        flowWorked = False

if(flowWorked):

    #download the extracted files
    grrapi.Client(client.client_id).File("/fs/tsk/").GetFilesArchive().WriteToFile("/tmp/flow-test.zip")
```

**stack trace**

```
Traceback (most recent call last):
  File "/home/xxxxxxxx/PycharmProjects/xxxxxxx/https-broken.py", line 71, in <module>
    grrapi.Client(client.client_id).File("/fs/tsk/").GetFilesArchive().WriteToFile("/tmp/flow-test.zip")
  File "/usr/local/lib/python2.7/dist-packages/grr_api_client/vfs.py", line 125, in GetFilesArchive
    return self._context.SendStreamingRequest("GetVfsFilesArchive", args)
  File "/usr/local/lib/python2.7/dist-packages/grr_api_client/context.py", line 61, in SendStreamingRequest
    return self.connector.SendStreamingRequest(handler_name, args)
  File "/usr/local/lib/python2.7/dist-packages/grr_api_client/connectors/http_connector.py", line 241, in SendStreamingRequest
    response = session.send(prepped_request, stream=True)
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 576, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/adapters.py", line 447, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: ("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')],)",)
```

**Comment by mbushkov** [comment]
Thank you for a thorough error report!
There was a bug in grr-api-client's code that handles streaming HTTP requests (GetFilesArchive is such a request). This bug is now fixed: https://github.com/google/grr/commit/217882d1748f4185af1b74a75bac8b9d0e0fa22e

I've updated grr-api-client package on PyPi: new version is 3.2.1.post6. Please try it, it should work correctly now.

By the way, Flow object has `WaitUntilDone` method that you can use instead of implementing your own waiting logic (https://github.com/google/grr/blob/master/api_client/python/grr_api_client/flow.py#L82). It waits until a flow completes and raises a `grr_api_client.errors.FlowFailedError` if a flow hasn't finished in time (max waiting time is set via the `timeout` argument with a default of one hour).

**Comment by soundwave01** [comment]
Thank you for the super fast response, and the tip on WaitUntilDone

I have tested the latest version

but I'm now getting 

`ParseError: Failed to parse items field: Failed to parse defaultArgs field: Can not find message descriptor by type_url: type.googleapis.com/CollectArtifactDependenciesArgs..`

on this call

`flow_args = grrapi.types.CreateFlowArgs("ArtifactFilesDownloaderFlow")`

Last time I had this was when I was using the 3.2.1 api code on a server running 3.2.0.1  However both my production and test grr setup are now running 3.2.1.1 (which solved the problem)

I'm not use if it's a dependency issues on my machine or not, I have removed all the grr pip packages and reinstalled them.  I still have the same problem.

Any idea what the issue is?

Thank you again for the all help on this!
